### PR TITLE
[19490] Fix Github windows CI

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR adds logic to the Windows CI workflow to cope with versions depending on different Fast CDR versions.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A*: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A*: Any new/modified methods have been properly documented using Doxygen. 
- *N/A*: Changes are ABI compatible. 
- *N/A*: Changes are API compatible. 
- *N/A*: New feature has been added to the `versions.md` file (if applicable).
- *N/A*: New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A*: Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/config/default_ci_2.6.x.repos [hunk 1]
```diff
@@ -0,0 +1,13 @@
+repositories:
+    fastcdr:
+        type: git
+        url: https://github.com/eProsima/Fast-CDR.git
+        version: 1.1.x
```
Comment (MiguelCompany) [inline .github/workflows/config/default_ci_2.6.x.repos:5]:
```suggestion
        version: 1.0.x
```

File: .github/workflows/config/default_ci_2.6.x.repos [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    foonathan_memory_vendor:
+        type: git
+        url: https://github.com/eProsima/foonathan_memory_vendor.git
+        version: master
```
Comment (MiguelCompany) [inline .github/workflows/config/default_ci_2.6.x.repos:9]:
```suggestion
        version: v1.2.1
```

File: .github/workflows/config/nightly_2.10.x.repos [hunk 3]
```diff
@@ -0,0 +1,13 @@
+repositories:
+    fastcdr:
+        type: git
+        url: https://github.com/eProsima/Fast-CDR.git
+        version: 1.1.x
```
Comment (MiguelCompany) [inline .github/workflows/config/nightly_2.10.x.repos:5]:
```suggestion
        version: 1.0.x
```

File: .github/workflows/config/nightly_2.10.x.repos [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    foonathan_memory_vendor:
+        type: git
+        url: https://github.com/eProsima/foonathan_memory_vendor.git
+        version: master
```
Comment (MiguelCompany) [inline .github/workflows/config/nightly_2.10.x.repos:9]:
```suggestion
        version: v1.3.0
```

File: .github/workflows/config/nightly_2.11.x.repos [hunk 5]
```diff
@@ -0,0 +1,13 @@
+repositories:
+    fastcdr:
+        type: git
+        url: https://github.com/eProsima/Fast-CDR.git
+        version: 1.1.x
+    foonathan_memory_vendor:
+        type: git
+        url: https://github.com/eProsima/foonathan_memory_vendor.git
+        version: master
```
Comment (MiguelCompany) [inline .github/workflows/config/nightly_2.11.x.repos:9]: v1.3.1

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
The suggestions below should also be applied to the nightly_xxx repos

**Review by MiguelCompany — APPROVED** [review]

**Comment by EduPonz** [comment]
@mergifyio backport 2.11.x 2.10.x 2.6.x

**Pull request merged**